src/config.rs

```diff
@@ -28,6 +27,85 @@ pub enum ConfigError {
     AmbiguousSource(PathBuf, PathBuf),
 }
 
+/// Set of configs which can be merged as needed.
+///
+/// Sources from the lowest precedence:
+/// 1. Default
+/// 2. Base environment variables
+/// 3. User config `~/.jjconfig.toml` or `$JJ_CONFIG`
+/// 4. Repo config `.jj/repo/config.toml`
+/// 5. TODO: Workspace config `.jj/config.toml`
+/// 6. Override environment variables
+/// 7. Command-line arguments `--config-toml`
```

Comment:
I think ordering it from highest to lowest makes it easier to read, it would also be nice to mention the user config path for windows users.

Reply:
> I think ordering it from highest to lowest makes it easier to read,

It might be odd as a precedence list, but is easier to compare with the implementation. So I prefer the current form.

> it would also be nice to mention the user config path for windows users

Oh, I thought it would be the same for Windows. Is `%APPDATA%/jj/config.toml` the canonical config path on Windows?

Reply:
I think it's either `%HOME%\.jjconfig.toml` or `C:\Users\%USER%\AppData\Roaming\jj\config.toml` (based on what I found on https://crates.io/crates/dirs), but I'm not sure at all about that.

Reply:
Yes, and 96849da332c8 says `~/.jjconfig.toml` is preferred. I have no idea which one is more friendly for Windows users, but that's the current state afaik.

I'll send another PR to update docs and this doc comment if AppData should be preferred.